android-components/components/browser/state/src/main/java/mozilla/components/browser/state/action/BrowserAction.kt

```diff
@@ -1102,6 +1102,13 @@ sealed class EngineAction : BrowserAction() {
         val throwable: Throwable,
     ) : EngineAction(), ActionWithTab
 
+    /**
+     * State of save to pdf error snackBar has changed
+     */
+    data class ShowSaveToPDFErrorSnackBarChange(
```

Comment:
```suggestion
    data class ShowStandardSnackbarErrorAction(
```
I think this could be useful to have around as a generic component. What do you think? 

Could you add a parameter of `val errorMsg: String` to the signature and rename this and the associated functions to something like `Standard` error?

(I plan on reusing this UI element for print and I expect it could be useful for other unrecoverable errors too. It would be great to call something like: `ShowStandardSnackbarErrorAction(true, R.string.unable_to_print_error)`)

Reply:
Yeah, completely agree, we will need to create an state that is generic enough that other similar use cases could use, as this is something that maybe app specific in this case Fenix we could put it under [AppState](https://github.com/mozilla-mobile/firefox-android/blob/main/fenix/app/src/main/java/org/mozilla/fenix/components/appstate/AppState.kt), also tagging @jonalmeida in case he could have other input :)

Reply:
This sounds like a good idea!

> Could you add a parameter of val errorMsg: String to the signature and rename this and the associated functions to something like Standard error?

Consider how this message coming from the component is going to be localized. I'm not sure yet, what's the best direction for this if it was the `errorMessage` that we display.

> this is something that maybe app specific in this case Fenix we could put it under [AppState](https://github.com/mozilla-mobile/firefox-android/blob/main/fenix/app/src/main/java/org/mozilla/fenix/components/appstate/AppState.kt),

This might not work. If we are dispatching an action from the BrowserStore at the engine-level we would have have to chain it to the AppStore before we can get it to the UI. This _might_ make it better for the point I mentioned above, but again I'm not sure, because now we are daisy-chaining two stores together.

Reply:
Done

Reply:
Thanks for the update! I tested by changing system languages and the message correctly localized.